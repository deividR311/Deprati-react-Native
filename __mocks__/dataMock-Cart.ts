export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
