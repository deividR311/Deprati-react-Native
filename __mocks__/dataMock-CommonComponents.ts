export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
