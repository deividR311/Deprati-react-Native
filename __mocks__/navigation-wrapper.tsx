import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
