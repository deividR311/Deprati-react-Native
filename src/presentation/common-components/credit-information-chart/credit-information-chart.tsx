import React from 'react';
import { Text, View } from 'react-native';
import { CreditInformationChartProps } from './interface';
import { Styles } from './credit-information-chart.stylesheet';
import { DonutChart } from './donut-chart';
import { FontStyles } from '../../../application/common/fonts';
import { capitalize } from '../../../application/utils';

export const CreditInformationChart: React.FC<
  CreditInformationChartProps
> = props => {
  const {
    constentChart,
    ownerFullname = '',
    numeroTarjetaAdicionalDisplay = '',
    affiliateDate = '',
    totalAmount = 1,
    amountSpent = 0,
    availableAmount = 0,
    showAffiliateDate = true,
    showDisplayCard = true,
    displayCartType = 'small',
    invertValues = false,
    wrapperChartInCard = false
  } = props;

  return (
    <View style={Styles.container}>
      <View
        style={[
          Styles.textContainer,
          displayCartType === 'large' ? Styles.largeContainerCard : undefined
        ]}>
        {showDisplayCard && displayCartType === 'small' && (
          <View style={Styles.leftTextContainer}>
            <Text style={[FontStyles.Caption]}>{ownerFullname}</Text>
            <Text style={[FontStyles.Caption]}>
              {numeroTarjetaAdicionalDisplay}
            </Text>
          </View>
        )}
        {showAffiliateDate && displayCartType === 'small' && (
          <Text style={[FontStyles.Caption, FontStyles.MutedColor]}>
            Cliente desde {affiliateDate}
          </Text>
        )}
        {showDisplayCard && displayCartType === 'large' && (
          <View style={[Styles.leftTextContainerLarge]}>
            <Text style={Styles.largeTitle}>Datos del cliente</Text>
            <View style={Styles.tableRow}>
              <Text style={[Styles.cellCaption]}>Nombre:</Text>
              <Text numberOfLines={1} style={[Styles.cellValue]}>
                {ownerFullname
                  ?.split(' ')
                  .map(n => capitalize(n, false))
                  .join(' ')}
              </Text>
            </View>
            <View style={Styles.tableRow}>
              <Text style={[Styles.cellCaption]}>CÃ³digo del cliente:</Text>
              <Text style={[Styles.cellValue]}>
                {numeroTarjetaAdicionalDisplay}
              </Text>
            </View>
            <View style={Styles.tableRow}>
              <Text style={[Styles.cellCaption]}>Cliente desde:</Text>
              <Text style={[Styles.cellValue]}>{affiliateDate}</Text>
            </View>
          </View>
        )}
      </View>
      <View
        style={[
          Styles.chartContainer,
          wrapperChartInCard ? Styles.wrapperChartInCard : undefined
        ]}>
        <View style={Styles.graphicContainer}>
          <DonutChart
            invertValues={invertValues}
            radius={107}
            maxIndicatorValue={totalAmount}
            indicatorValue={amountSpent}
          />
        </View>
        <View>
          {!invertValues && (
            <>
              <Text
                style={[
                  FontStyles.Body_2,
                  FontStyles.MutedColor,
                  FontStyles.Center
                ]}>
                Cupo utilizado
              </Text>
              <Text
                style={[
                  FontStyles.H1_Headline,
                  FontStyles.PrimaryColor,
                  { marginBottom: 8 }
                ]}>
                ${constentChart?.amountSpent ?? '0,00'}
              </Text>
            </>
          )}

          <Text
            style={[
              FontStyles.Body_2,
              invertValues ? FontStyles.Bold : undefined,
              invertValues ? FontStyles.DarkColor : FontStyles.MutedColor,
              FontStyles.Center,
              invertValues ? { marginTop: 8 } : undefined
            ]}>
            Cupo disponible
          </Text>
          <Text
            style={[
              invertValues ? FontStyles.H1_Headline : FontStyles.Body_2,
              FontStyles.Center,
              { marginBottom: 6 }
            ]}>
            ${constentChart?.availableAmount ?? '0,00'}
          </Text>

          {invertValues && (
            <>
              <Text
                style={[
                  FontStyles.Body_2,
                  FontStyles.MutedColor,
                  FontStyles.Center
                ]}>
                Cupo utilizado
              </Text>
              <Text style={[FontStyles.Body_2, FontStyles.Center]}>
                ${constentChart?.amountSpent ?? '0,00'}
              </Text>
            </>
          )}

          <Text
            style={[
              FontStyles.Body_2,
              FontStyles.MutedColor,
              FontStyles.Center
            ]}>
            Cupo autorizado
          </Text>
          <Text style={[FontStyles.Body_2, FontStyles.Center]}>
            ${constentChart?.totalAmount ?? '0,00'}
          </Text>
        </View>
      </View>
    </View>
  );
};
