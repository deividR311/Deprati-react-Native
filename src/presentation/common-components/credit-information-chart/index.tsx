export * from './credit-information-chart';
export * from './donut-chart';
export * from './interface';
