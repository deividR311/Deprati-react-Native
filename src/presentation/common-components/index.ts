import BannerHome from './banner-home';
import CategoryHome from './category-home';
import StoriesHome from './stories-home';
import CarouselWelcome from '../screens/Welcome/components/CarouselWelcome';
import CategoryMain from '../screens/CategoryPage/components/CategoryMain';
import CategoryFeature from '../screens/CategoryPage/components/CategoryFeature';
import BannerCarousel from '../screens/CategoryPage/components/BannerCarousel';
import RecommendedProducts from './recommended-home/RecommendProducts';
import SimpleBanner from './banner-home/SimpleBanner';
import CMSParagraph from './CMSParagraph/CMSParagraph';
import CMSLinkButton from './CMSLinkButton/CMSLinkButton';
import DepratiBrandsRotatingImages from './DepratiBrandsRotatingImages/DepratiBrandsRotatingImages';
import BannerComponent from './banner-home/BannerComponent';
import DepratiDoubleResponsiveBanner from './DepratiDoubleResponsiveBanner/DepratiDoubleResponsiveBannerComponent';
import DepratiHomeRotatingImagesComponent from './DepratiHomeRotatingImagesComponent/DepratiHomeRotatingImagesComponent';
import { DepratiSimpleResponsiveBannerComponent } from '../screens/dashboard/PDP/components/DepratiSimpleResponsiveBannerComponent';
import ProductCarouselComponent from './productCarouselComponent';
import DepratiCloudflareVideoStreamBannerComponent from './DepratiCloudflareVideoStreamBannerComponent/DepratiCloudflareVideoStreamBannerComponent';

interface ComponentType {
  [key: string]: React.ReactNode;
}

// uid : Component
export const Components: ComponentType = {
  //Home component
  DepratiStoryContainerComponent: StoriesHome,
  SimpleBannerComponent: SimpleBanner,
  DepratiCategoryListComponent: CategoryHome,
  DepratiEmarsysProductCarouselComponent: RecommendedProducts,
  DepratiHomeRotatingImagesComponent: DepratiHomeRotatingImagesComponent,

  //Welcome Component
  DepratiWelcomeRotatingImagesComponent: CarouselWelcome,
  RotatingImagesComponent: CarouselWelcome,

  //Category Component
  DepratiCategoryFeatureCarouselComponent: CategoryMain,
  CategoryFeatureComponent: CategoryFeature,
  DepratiDoubleResponsiveBannerComponent: DepratiDoubleResponsiveBanner, //BannerCarousel
  DepratiSimpleResponsiveBannerComponent,
  CMSParagraphComponent: CMSParagraph,
  CMSLinkComponent: CMSLinkButton,
  DepratiBrandsRotatingImagesComponent: DepratiBrandsRotatingImages,
  BannerComponent: BannerComponent,
  DepratiCloudflareVideoStreamBannerComponent,
  SimpleResponsiveBannerComponent: SimpleBanner,
  //Corner
  ProductCarouselComponent
};
