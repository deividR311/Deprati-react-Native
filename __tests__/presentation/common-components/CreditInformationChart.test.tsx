import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../__mocks__/navigation-wrapper';
import { CreditInformationChart } from '../../../src/presentation/common-components/credit-information-chart';
import { DataMock_Charts } from '../../../__mocks__/dataMock-CommonComponents';

let screenTest: any;
describe('CreditInformationChart', () => {
  it('renders correctly CreditInformationChart in Snapshot', () => {
    screenTest = render(
      <NavigationWrapper
        screen={() => (
          <CreditInformationChart
            constentChart={DataMock_Charts.constentChart}
          />
        )}
      />
    );
    expect(screenTest.toJSON()).toMatchSnapshot();
  });

  it('renders alternative params', () => {
    screenTest = render(
      <NavigationWrapper
        screen={() => (
          <CreditInformationChart
            {...DataMock_Charts}
            displayCartType="large"
          />
        )}
      />
    );
    expect(screenTest.queryByText(DataMock_Charts.ownerFullname)).toBeDefined();
  });
});
