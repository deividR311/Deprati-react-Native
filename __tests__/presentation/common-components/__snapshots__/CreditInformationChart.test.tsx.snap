// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditInformationChart renders correctly CreditInformationChart in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0.8,
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0.8,
                    },
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "letterSpacing": 0.8,
                  },
                  Object {
                    "color": "#26323880",
                  },
                ]
              }
            >
              Cliente desde 
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingTop": 35,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 214,
                    "width": 214,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={214}
                  bbWidth={214}
                  fill="gray"
                  focusable={false}
                  height={214}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 214,
                        "width": 214,
                      },
                    ]
                  }
                  vbHeight={242}
                  vbWidth={242}
                  width={214}
                >
                  <RNSVGGroup
                    fill={4286611584}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup
                      matrix={
                        Array [
                          1,
                          0,
                          -0,
                          1,
                          0,
                          0,
                        ]
                      }
                    >
                      <RNSVGCircle
                        cx="50%"
                        cy="50%"
                        fill={0}
                        matrix={
                          Array [
                            -0.8660254037844386,
                            0.5000000000000001,
                            -0.5000000000000001,
                            -0.8660254037844386,
                            286.28907385791706,
                            165.28907385791706,
                          ]
                        }
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeWidth",
                            "strokeDasharray",
                            "strokeDashoffset",
                            "strokeLinejoin",
                          ]
                        }
                        r={107}
                        stroke={4293322470}
                        strokeDasharray={
                          Array [
                            672.3008278682157,
                            672.3008278682157,
                          ]
                        }
                        strokeDashoffset={224.1002759560719}
                        strokeLinejoin={1}
                        strokeWidth={14}
                      />
                      <RNSVGCircle
                        cx="50%"
                        cy="50%"
                        fill={0}
                        matrix={
                          Array [
                            -0.8660254037844386,
                            0.5000000000000001,
                            -0.5000000000000001,
                            -0.8660254037844386,
                            286.28907385791706,
                            165.28907385791706,
                          ]
                        }
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeWidth",
                            "strokeDasharray",
                            "strokeDashoffset",
                            "strokeLinejoin",
                          ]
                        }
                        r={107}
                        stroke={4293073452}
                        strokeDasharray={
                          Array [
                            672.3008278682157,
                            672.3008278682157,
                          ]
                        }
                        strokeDashoffset={224.1002759560719}
                        strokeLinejoin={1}
                        strokeWidth={14}
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "color": "#26323880",
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Cupo utilizado
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 23,
                      "fontWeight": "700",
                      "letterSpacing": 0.44,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#E31A2C",
                    },
                    Object {
                      "marginBottom": 8,
                    },
                  ]
                }
              >
                $
                20
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    undefined,
                    Object {
                      "color": "#26323880",
                    },
                    Object {
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Cupo disponible
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "center",
                    },
                    Object {
                      "marginBottom": 6,
                    },
                  ]
                }
              >
                $
                100
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "color": "#26323880",
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Cupo autorizado
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                $
                200
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
